### DISCUSSION TITLE
ADIF values

### DISCUSSION THREAD

**Discussion by 2e0wkw** [magicbug/Cloudlog#1604] [General]
I am not sure if this is the right place to ask this but...
I am trying to POST some ADIF data into index,php/API/QSO but i am getting errors with the adit properties as below

please excuse the html but node red shows it as text

<h4>A PHP Error was encountered</h4>

<p>Severity: Notice</p>
<p>Message:  Undefined index: qso_date</p>
<p>Filename: models/Logbook_model.php</p>
<p>Line Number: 1645</p>

</div><div style="border:1px solid #990000;padding-left:20px;margin:0 0 10px 0;">

<h4>A PHP Error was encountered</h4>

<p>Severity: Notice</p>
<p>Message:  Undefined index: time_on</p>
<p>Filename: models/Logbook_model.php</p>
<p>Line Number: 1645</p>

</div><div style="border:1px solid #990000;padding-left:20px;margin:0 0 10px 0;">

<h4>A PHP Error was encountered</h4>

<p>Severity: Notice</p>
<p>Message:  Undefined index: call</p>
<p>Filename: models/Logbook_model.php</p>
<p>Line Number: 2173</p>

</div>{"status":"created","type":"adif","string":"<adif ver:5>3.1.3 <programid:17>mac smart sdr mac os <eoh> <call:7>dl1 test<qso date:8>20220914<time on:6>163453<time off:6>163453<band:3>20 m<freq:6>14.075<mode:3>ft8<rst sent:2> 1<rst rcvd:2>..

**Comment by magicbug** [comment]
Would need to see what the adif string is to know for sure.

**Reply by 2e0wkw** [reply]
I am sending 
"{"key":"HIDDEN","station_profile_id":3,"type":"adif","string":"<adif ver:5>3.1.3 <programid:17>mac smart sdr mac os <eoh> <call:7>dl1 test<qso date:8>20220914<time on:6>165528<time off:6>165528<band:3>20 m<freq:6>14.075<mode:3>ft8<rst sent:2> 1<rst rcvd:2> 2<comment:19>udp test submission<eor>"}"

**Reply by phl0** [reply]
That string does not look like ADIF ...

**Reply by 2e0wkw** [reply]
looks like the forum removed the '<' and '>'

**Reply by phl0** [reply]
Post it as quote to make sure.

**Reply by 2e0wkw** [reply]
`"{"key":"HIDDEN","station_profile_id":3,"type":"adif","string":"<adif ver:5>3.1.3 <programid:17>mac smart sdr mac os <eoh> <call:7>dl1 test<qso date:8>20220914<time on:6>165528<time off:6>165528<band:3>20 m<freq:6>14.075<mode:3>ft8<rst sent:2> 1<rst rcvd:2> 2<comment:19>udp test submission<eor>"}"`

**Reply by AndreasK79** [reply]
The tags are wrong. qso date must be qso_date. Time on must be time_on. Time off must be time_off. rst sent must be rst_sent and rst rcvd must be rst_rcvd.

**Reply by phl0** [reply]
And call:7 is also wrong.

**Reply by phl0** [reply]
![Screenshot from 2022-09-14 18-15-44](https://user-images.githubusercontent.com/7112907/190207744-052d61c0-0000-4e3b-a24e-0902e719c000.png)

**Reply by 2e0wkw** [reply]
am i missing 'country' ?

**Reply by magicbug** [reply]
This is a simple case of incorrectly formatted ADIF as has already been said. The country can be missing Cloudlog will figure it out if need be.

If you are unsure about the ADIF standard you can find out more at [https://adif.org/](https://adif.org/)

**Reply by 2e0wkw** [reply]
OK i am making progress with your help - thanks
I am missing their callsign in the log now.
This is my latest POST
`"{"key":"HIDDEN","station_profile_id":3,"type":"adif","string":"<adif_ver:5>3.1.3 <programid:17>MacSmartSDR-macOS <EOH> **<CALL:5>G4NMD<QSO_DATE:8>20220914<TIME_ON:6>164439<TIME_OFF:6>164448<BAND:3>20M<FREQ:6>14.264<MODE:3>SSB<SUBMODE:3>USB<RST_SENT:2>59<RST_RCVD:2>59<OPERATOR:5>M0REQ<STATION_CALLSIGN:5>M0REQ<CONT:2>EU<COUNTRY:7>England<DXCC:3>223<CQZ:2>14<ITUZ:2>27<LAT:7>52.7778<LON:7>-1.5556<EOR>"}"`

I am not sure what is wrong with <CALL:5>G4NMD

I am getting the following error message:
`
<h4>A PHP Error was encountered</h4>

<p>Severity: Notice</p>
<p>Message:  Trying to access array offset on value of type null</p>
<p>Filename: models/Logbook_model.php</p>
<p>Line Number: 2018</p>

</div>{"status":"created","type":"adif","string":"<adif_ver:5>3.1.3 <programid:17>MacSmartSDR-macOS <EOH> <CALL:5>G4NMD<QSO_DATE:8>20220914<TIME_ON:6>164439<TIME_OFF:6>164448<BAND:3>20M<FREQ:6>14.264<MODE:3>SSB<SUBMODE:3>USB<RST_SENT:2>59<RST_RCVD:2>59<OPERATOR:5>M0REQ<STATION_CALLSIGN:5>M0REQ<CONT:2>EU<`

**Reply by magicbug** [reply]
`<LAT:7>52.7778<LON:7>-1.5556` are incorrectly formatted 

http://hamclubs.info/adif-validator/
